authorization

annkonst opened the pull request.

reshetov commented on app/controllers/lessons_controller.rb:
```diff
@@ -1,6 +1,8 @@
 class LessonsController < ApplicationController
+  before_filter :authenticate_user!
```
У тебя это уже есть в application_controller. Зачем еще раз?

reshetov commented on app/controllers/lessons_controller.rb:
```diff
@@ -9,9 +11,10 @@ def index
     @lesson = Lesson.where(id: params[:id]).first
     return redirect_to(lessons_path, flash[:error] = t(:error_show)) unless @lesson
-    @users = User.where(group_id: @lesson.group_id).includes(:lesson_marks)
+    @users = User.where(group_id: @lesson.group_id).where.not(name: 'Ksenia').includes(:lesson_marks)
```
Что это?

reshetov commented on app/models/user.rb:
```diff
@@ -1,7 +1,11 @@
   has_many :lesson_marks
   has_many :lessons, through: :lesson_marks
-  validates :group_id, :name, presence: true
+  validates :group_id, :name,  presence: true
```
пробельчик

Commit:
report with scroll

Commit:
report

Commit:
# This is a combination of 2 commits.
# The first commit's message is:

report

# The 2nd commit message will be skipped:

#	report

Commit:
Added Gemfile.lock

Commit:
Merge branch 'master' into authorization

Conflicts:
	.gitignore
	app/assets/stylesheets/application.sass
	app/views/devise/registrations/new.html.slim
	app/views/devise/sessions/new.html.slim
	app/views/layouts/application.html.slim
	app/views/lessons/_show.html.slim
	app/views/lessons/index.html.slim

Commit:
change .gitignore

Commit:
report for heroku

reshetov commented on app/controllers/documents_controller.rb:
```diff
@@ -0,0 +1,8 @@
+class DocumentsController < ApplicationController
+
+  def destroy
+    Document.find(params[:id]).destroy
```
если не найдет шлепнется с `exception`

reshetov commented on app/controllers/lessons_controller.rb:
```diff
@@ -38,14 +43,40 @@ def update
+    docs = params[:documents]
+    if docs.present?
+      if docs.is_a? Array
+        docs.each { |doc| lesson.documents.create file: doc}
```
пробел перед  `}`

reshetov commented on app/controllers/lessons_controller.rb:
```diff
@@ -38,14 +43,40 @@ def update
+
+  def destroy_all_documents
+    @lesson = Lesson.find(params[:id])
+    @lesson.documents.each do |del_doc|
```
а если не найдет?

reshetov commented on app/controllers/lessons_controller.rb:
```diff
@@ -18,11 +18,16 @@ def show
     @lesson = Lesson.new(lesson_params)
-    flash[:error] = t(:error_create) unless @lesson.save
+    if @lesson.save
+      create_document(@lesson)
```
Еще раз вопрос этот метод точно нужен? У тебя есть accept_nested_attributes. Их должно хватить для создания документов автоматом

annkonst replied:
точно нужен

reshetov commented on app/views/devise/shared/_links.html.erb:
```diff
@@ -6,9 +6,9 @@
-<%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
-  <%= link_to "Forgot your password?", new_password_path(resource_name) %><br />
-<% end -%>
+<!--- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>-->
```
???

reshetov commented on app/views/lessons/_show.html.slim:
```diff
@@ -15,7 +15,17 @@
+            - @lesson.documents.each do |doc|
+              li
+                = link_to doc.file_file_name, doc.file.url
+                = ' / '
```
Зачем там сложно то?
|/

Commit:
report with corrected delete method
